rsconf/component/__init__.py

```diff
@@ -157,7 +210,7 @@ def install_resource(self, name, j2_ctx, host_path=None):
             host_path = name
             if host_path.ext == ".sh":
                 host_path = host_path.new(ext="")
-            name = self.name + "/" + name.basename
+            name = (self.get("module_name") or self.name) + "/" + name.basename
```

Comment:
I think there should always be a module_name. For backward compatibility, you will need to only do this if "/" isn't already in name iwc the caller already added module_name (typical case). This is a highly constrained problem, because we never have a "name" without a "/" (you can assert this), that is, all resources are at least two levels deep. Therefore I think this trick will be sufficient to fix all cases.

Reply:
> you will need to only do this if "/" isn't already in name

I'm not sure what you mean by "this"?

I think just using self.module_name works 749c4cb. We default module_name to name. That I believe satisfies our backwards compat needs. In the devbox case module_name is devbox (not devbox_<username> which is what self.name is) so it works in that case as well.